### ISSUE TITLE
buttons - z-index and waves-effect

### ISSUE THREAD

**Issue by opb** [Dogfalo/materialize#1048]
Hi there

I have come across a couple of issues with recent `btn` css styling. 

1. If you include `btn` as the only class on an `a` tag (for example: `<a href="#" class="btn">My Button</a>`) then the css on the `.btn:before:hover` goes very strange and instead of providing a shadow to the button, turns the container grey:

![screen shot 2015-03-30 at 15 08 54](https://cloud.githubusercontent.com/assets/778850/6898148/332d3d46-d6f0-11e4-8cca-a3014a669722.png)

If you add `waves-effect` to the class, this effect is removed.

2. On buttons where `waves-effect` is included, these have a lower z-index than the menus:

![screen shot 2015-03-30 at 15 00 52](https://cloud.githubusercontent.com/assets/778850/6898191/780ed91a-d6f0-11e4-8b84-65b32d826079.png)

Please let me know if you need any further information.

**Comment by acburst** [comment]
Can you post example code for the first issue?

**Comment by opb** [comment]
Hi there

I have worked out what causes the problem in the first example.

The library styling applied to `.btn:before` includes `position: absolute; width: 100%; height: 100%`. However, the button receives its `position: relative` from the `.waves-effect` styling. 

Without that class, the `.btn` has no `position` and so the `absolute` styling is relative to the next element up with a `position` - in our case this is a content wrapper.

So basically, the 100% height and width `.btn:before` is not constrained to our `.btn` because we don't have the `.waves-effect` applied to our button, just the `btn` class.

**Comment by Dogfalo** [comment]
We were testing out some new ways to retain hover color in buttons, but we've reverted to the old style. If you can confirm that this no longer happens, we will close this issue. Thanks.

**Comment by opb** [comment]
Confirmed, I was using `master` on another project yesterday and all seems fine. Many thanks!
